lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java

```diff
@@ -0,0 +1,131 @@
+  private static final Logger log = LogManager.getLogger(TextExtractor.class);
+  private String textField;
+  private String filePathField;
+  private String tikaConfigPath;
+  private String byteArrayField;
+  private String metadataPrefix;
+  private Parser parser;
+  private ParseContext parseCtx;
+
+  public TextExtractor(Config config) {
+    super(config, new StageSpec().withOptionalProperties("text_field", "file_path_field", "byte_array_field", "tika_config_path",
+        "metadata_prefix"));
+    textField = config.hasPath("text_field") ? config.getString("text_field") : "text";
+    filePathField = config.hasPath("file_path_field") ? config.getString("file_path_field") : "filepath";
+    byteArrayField = config.hasPath("byte_array_field") ? config.getString("byte_array_field") : "byte_array";
+    metadataPrefix = config.hasPath("metadata_prefix") ? config.getString("metadata_prefix") : "tika";
+    tikaConfigPath = config.hasPath("tika_config_path") ? config.getString("tika_config_path") : null;
+    parseCtx = new ParseContext();
+  }
+
+  @Override
+  public void start() throws StageException {
+    if (this.tikaConfigPath == null) {
+      parser = new AutoDetectParser();
+    } else {
+      try {
+        File f = new File(this.tikaConfigPath);
+        TikaConfig tc = new TikaConfig(f);
+        parser = new AutoDetectParser(tc);
+      } catch (Exception e) {
+        throw new StageException("Error starting TextExtractor stage.", e);
+      }
+    }
+    parseCtx.set(Parser.class, parser);
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) throws StageException {
+    // if the document has both a byteArray field and a filePathField, only byteArray will be processed.
+    if (doc.has(byteArrayField)) {
+
+      byte[] byteArray = doc.getBytes(byteArrayField);
+
+      InputStream inputStream = new ByteArrayInputStream(byteArray);
+      parseInputStream(doc, inputStream);
+
+    } else if (doc.has(filePathField)) {
+
+      String filePath = doc.getString(filePathField);
+
+      try {
+        InputStream inputStream = FileUtils.getInputStream(filePath);
+        parseInputStream(doc, inputStream);
+      } catch (IOException e) {
+        throw new StageException("InputStream cannot be parsed or created", e);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Cleans the name of metadata field names to be in line with general standards for documents
+   */
+  private static String cleanFieldName(String name) {
+    String cleanName = name.trim().toLowerCase();
+    cleanName = cleanName.replaceAll(" ", "_");
+    cleanName = cleanName.replaceAll("-", "_");
+    cleanName = cleanName.replaceAll(":", "_");
+    return cleanName;
+  }
+
+  /**
+   * Parses given input stream and adds the text data and metadata to given document
+   */
+  public void parseInputStream(Document doc, InputStream inputStream) {
+    Metadata metadata = new Metadata();
+    ContentHandler bch = new BodyContentHandler();
```

Comment:
IIRC the BodyContentHandler has a constructor that takes a numeric value that allows you to limit the size of the content that it will produce. We should allow this value to be configurable.

Reply:
Changed this in new commit, should now take in config property to limit size (there is also a test for this as well now)